rct229/rules/section6/section6rule8.py

```diff
@@ -0,0 +1,128 @@
+                        each_rule=Section6Rule8.RuleSetModelInstanceRule.BuildingRule.ZoneRule.SpaceRule(),
+                        index_rmr="proposed",
+                        list_path="spaces[*]",
+                        required_fields={
+                            "$": ["volume"]
+                        }
+                    )
+
+                def create_data(self, context, data=None):
+                    zone_p = context.proposed
+                    return {
+                        **data,
+                        "avg_space_height": zone_p.get("volume", 0.0) / pint_sum(find_all("spaces[*].floor_area", zone_p), ZERO.AREA)
+                    }
+
+                class SpaceRule(RuleDefinitionBase):
+                    def __init__(self):
+                        super(Section6Rule8.RuleSetModelInstanceRule.BuildingRule.ZoneRule.SpaceRule, self).__init__(
+                            rmrs_used=UserBaselineProposedVals(False, True, True)
+                        )
+
+                    def get_calc_vals(self, context, data=None):
+                        space_p = context.proposed
+                        space_b = context.baseline
+                        avg_space_height = data["avg_space_height"]
+                        schedules_b = data["schedules_b"]
+                        schedules_p = data["schedules_p"]
+                        building_open_schedule_p = data["building_open_schedule_p"]
+
+                        normalized_schedule_b = normalize_interior_lighting_schedules(space_b, avg_space_height, schedules_b)
+                        normalized_schedule_p = normalize_interior_lighting_schedules(space_p, avg_space_height, schedules_p)
+                        schedule_comparison_result = compare_schedules(normalized_schedule_p, normalized_schedule_b, building_open_schedule_p, 1.0)
+
+                        return {
+                            "total_hours_compared": schedule_comparison_result["total_hours_compared"],
+                            "total_hours_matched": schedule_comparison_result["total_hours_compared"],
+                        }
+
+                    def manual_check_required(self, context, calc_vals=None, data=None):
+                        total_hours_compared = calc_vals["total_hours_compared"]
+                        total_hours_matched = calc_vals["total_hours_matched"]
+                        return total_hours_matched > total_hours_compared
+
+                    def get_manual_check_required_msg(self, context, calc_vals=None, data=None):
+                        return CASE_3_MSG
+
+                    def rule_check(self, context, calc_vals=None, data=None):
+                        total_hours_compared = calc_vals["total_hours_compared"]
+                        total_hours_matched = calc_vals["total_hours_matched"]
+                        return total_hours_matched == total_hours_compared
+
+                    def get_fail_msg(self, context, calc_vals=None, data=None):
```

Comment:
Similar to the manual check required, this could be replaced by adding `fail_msg=CASE2_MSG`. And I suggest calling it `FAIL_MSG` instead of `CASE_2_MSG`.